source/shared/cpp/ObjectModel/BaseActionElement.cpp

```diff
@@ -132,4 +147,5 @@ void BaseActionElement::DeserializeBaseProperties(ParseContext& context, const J
     element->SetTitle(ParseUtil::GetString(json, AdaptiveCardSchemaKey::Title));
     element->SetIconUrl(ParseUtil::GetString(json, AdaptiveCardSchemaKey::IconUrl));
     element->SetStyle(ParseUtil::GetString(json, AdaptiveCardSchemaKey::Style, defaultStyle, false));
+    element->SetMode( ParseUtil::GetEnumValue<Mode>(json, AdaptiveCardSchemaKey::Mode,Mode::Primary, ModeFromString));
 }
```

Comment:
3 more places need to update in `BaseActionElement.cpp`:

1. constructor needs to set default value:
```
BaseActionElement::BaseActionElement(ActionType type) : 
  m_style(BaseActionElement::defaultStyle),
  m_type(type),
  m_mode(Mode::Primary)
```

2. `SerializeToJsonValue()` needs to update. Add
```
    if (m_mode != Mode::Primary)
    {
        root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Mode)] = ModeToString(m_mode);
    }
```

3. `PopulateKnownPropertiesSet()` needs to update. Append this argument into `insert()`'s initializer list:
```
AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Mode)}
```

Reply:
Done